src/app/views/query-response/pivot-items/pivot-items.tsx

```diff
@@ -56,13 +57,14 @@ export const getPivotItems = () => {
       const { itemKey } = link;
       const itemKeyString: string = !itemKey ? '' : itemKey.toString();
       const target = findTarget(itemKeyString);
-      const targetStepIndex = getTargetStepIndex(target, item.key)
+      const targetStepIndex = getTargetStepIndex(target, item.key, tour.tourSteps, tour.beginner)
```

Comment:
This function `getTargetStepIndex` has 4 parameters. It looks like the function could be doing more than one thing. If so, it can be split into smaller functions and they can be called in succession as they resolve to create the parameters for the next function.

Another option would be to create an object and pass the object into the function if the function truly requires all the parameters.

Alternatively specifically for this case,  `tour.tourSteps` and `tour.beginner` can just be passed in as one and destructured within the function.